hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -711,12 +711,30 @@ public void rollbackInflightCompaction(HoodieInstant inflightInstant, HoodieTabl
     table.getActiveTimeline().revertCompactionInflightToRequested(inflightInstant);
   }
 
+  /**
+   * Get inflight time line exclude compaction and clustering.
+   * @param table
+   * @return
+   */
+  private HoodieTimeline getInflightTimelineExcludeCompactionAndClustering(HoodieTable<T, I, K, O> table) {
```

Comment:
@lw309637554 Can you add this API to the HoodieTimeline ? You can name it `filterPendingExcludingCompactionAndReplace` for now.

Reply:
@n3nash Hello, about this API in timeline or as a util have  discussed with @satishkotha . Finally decided to put it here. Because here just need to exclude clustering of replace, not insert_overwrite of replace. Also check clustering need use metaclient, but timeline do not have metaclient object.